### PR TITLE
[Windows][PT2.6] Split larger Other Kernels lib

### PR DESCRIPTION
**Pull request by ratnampa**
torch_xpu_ops_sycl_kernels leads to around 1.83GB in size on windows, splitting it to reduce the lib size.

New libs introduced in this PR:

torch_xpu_ops_sycl_tensor_srcs
torch_xpu_ops_sycl_norm_loss_srcs
torch_xpu_ops_sycl_poly_srcs
torch_xpu_ops_sycl_dist_srcs

### CODE REVIEW

File: src/BuildOnWindows.cmake [hunk 1]
```diff
@@ -21,7 +16,6 @@ add_library(
   ${ATen_XPU_NATIVE_CPP_SRCS}
   ${ATen_XPU_GEN_SRCS})
 install(TARGETS torch_xpu_ops_aten DESTINATION "${TORCH_INSTALL_LIB_DIR}")
-# target_compile_definitions(torch_xpu_ops_aten PRIVATE CAFFE2_BUILD_MAIN_LIB)
```
Comment (fengyuan14) [inline src/BuildOnWindows.cmake:24]: Could you elaborate why we needed it before and why we deprecate it now?

**Comment by fengyuan14** [comment]
Could you also add `xe2-hpg` to Windows AOT list?

File: src/BuildOnWindows.cmake [hunk 2]
```diff
@@ -48,8 +42,23 @@ else()
   set(ATen_XPU_SYCL_REDUCE_SRCS)
   set(ATen_XPU_SYCL_ACTIVATION_SRCS)
   set(ATen_XPU_SYCL_FOREACH_SRCS)
+  set(ATen_XPU_SYCL_OTHERS_SLOW_SRCS)
+  set(ATen_XPU_SYCL_OTHERS_FAST_SRCS)
   set(ATen_XPU_SYCL_OTHERS_SRCS)
-
+  list(APPEND slow_patterns
```
[the remark this answers is not in the document]
Comment (ratnampa) [reply src/BuildOnWindows.cmake:48]: Slow is just dependent on the obj file size, I have added large files from the other sycl kernels here, as I wasn't able to find any new patterns that can help split the libs further. I will take another look and will try to refactor.

File: src/BuildOnWindows.cmake (hunk 1 above)
Comment (ratnampa) [reply src/BuildOnWindows.cmake:24]: It was already commented out, so I removed it. It wasn't used before too.

**Comment by ratnampa** [comment]
> Could you also add `xe2-hpg` to Windows AOT list in the PR?

Sure, will add this one itself.

File: src/BuildOnWindows.cmake (hunk 2 above)
[the remark this answers is not in the document]
Comment (ratnampa) [reply src/BuildOnWindows.cmake]: I have changed the split libs, now the torch_xpu_ops_sycl_kernels.dll size is now down to 1.24GB

File: src/BuildOnWindows.cmake [hunk 3]
```diff
@@ -74,11 +79,19 @@ else()
...
+	elseif(NOT IS_NORM STREQUAL "" OR NOT IS_LOSS STREQUAL "" OR NOT IS_RESIZE STREQUAL "")
+	  list(APPEND ATen_XPU_SYCL_NORM_LOSS_SRCS ${sycl_src})
+	elseif(NOT IS_POLY STREQUAL "")
+	  list(APPEND ATen_XPU_SYCL_POLY_SRCS ${sycl_src})
     else()
       list(APPEND ATen_XPU_SYCL_OTHERS_SRCS ${sycl_src})
     endif()
   endforeach()
-
+
```
Comment (fengyuan14) [inline src/BuildOnWindows.cmake:94]:
```suggestion

```

File: src/BuildOnWindows.cmake [hunk 4]
```diff
@@ -48,8 +42,12 @@ else()
   set(ATen_XPU_SYCL_REDUCE_SRCS)
   set(ATen_XPU_SYCL_ACTIVATION_SRCS)
   set(ATen_XPU_SYCL_FOREACH_SRCS)
+  set(ATen_XPU_SYCL_TENSOR_SRCS)
+  set(ATen_XPU_SYCL_NORM_LOSS_SRCS)
+  set(ATen_XPU_SYCL_POLY_SRCS)
+  set(ATen_XPU_SYCL_DISTRIBUTION_SRCS)
   set(ATen_XPU_SYCL_OTHERS_SRCS)
-
+
```
Comment (fengyuan14) [inline src/BuildOnWindows.cmake:50]:
```suggestion

```

**Comment by fengyuan14** [comment]
LGTM, @ratnampa Could you list the sizes of all kernel libraries in the PR description?

**Review by fengyuan14 — APPROVED** [review]

**Pull request merged**